### PR TITLE
Coach report bug fixes - date ranging and translation

### LINKED ISSUE #5022: Coach reports: Apply date rage to tabular report
**Detailed description:**
For the date range selected total exercise attempted is shown as 7, but the tabular column shows all the exercise of the all the kids instead of the selected date range.

**Expected behavior:**
Only the 7 exercise should be displayed on the column so that its easy for the teachers to see the progress made by the kids on a selected date range.

**Actual behavior:**
All the exercise are getting displayed for all the kids even though the selected date range shows total attempted exercise as 7. This is a huge problem for the teacher who has to scroll through all the exercise and is very time consuming. 
screen shot 2016-03-21 at 6 44 31 pm
screen shot 2016-03-21 at 6 45 13 pm
screen shot 2016-03-21 at 6 44 50 pm
screen shot 2016-03-21 at 6 44 56 pm
screen shot 2016-03-21 at 6 45 03 pm

Please refer the screenshots for more details.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Ensures the end date is set properly for the tabular report. Fix a bug where already translated titles were being translated again on the client side, causing an error.

## TODO

If not all TODOs are marked, this PR is considered WIP (work in progress)

- [ ] Have **tests** been written for the new code? If you're fixing a bug, write a regression test (or have a really good reason for not writing one... and I mean **really** good!)
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file
- [ ] Add an entry to CHANGELOG.rst

## Reviewer guidance

*If you PR has a significant size, give the reviewer some helpful remarks*

## Issues addressed

Aimed to fix #5022 - but that is a result of generaterealdata, so don't expect it to fix it.

### CODE REVIEW

File: kalite/coachreports/api_views.py [hunk 1]
```diff
@@ -120,12 +120,12 @@ def learner_logs(request):
         LogModel, fields, id_field, obj_ids, objects = return_log_type_details(log_type, topic_ids)
 
         log_objects = LogModel.objects.filter(user__in=learners, **obj_ids).values(*fields)
+
         if not topic_ids:
             topic_objects = log_objects.filter(latest_activity_timestamp__gte=start_date, latest_activity_timestamp__lte=end_date)
-            if topic_objects.count() == 0:
-                topic_objects = log_objects
```
Comment (jamalex) [inline kalite/coachreports/api_views.py]: What happens, now, if it's empty?

File: kalite/coachreports/static/js/coachreports/coach_reports/models.js [hunk 2]
```diff
@@ -33,6 +33,18 @@ var GroupCollection = Backbone.Collection.extend({
     url: window.GROUP_RESOURCE_URL
 });
 
+function normalizeEndDate(end_date) {
```
Comment (jamalex) [inline kalite/coachreports/static/js/coachreports/coach_reports/models.js]: I think we were trying to use underscore-casing for everything. But not a blocker.

**Comment by jamalex** [comment]
A couple of questions/nitpicks, but looks good to me.

**Comment by jamalex** [comment]
(Merged so we can get it into the final 0.16.0 build.

**Comment by rtibbles** [comment]
Hrm, I wasn't happy with this because it broke tests :(

**Pull request merged**